Add CLI for cloudkeeper-aws

dudoslav opened the pull request:
CLI includes method sync and version. Also this commit adds THOR into
dependencies, gem bin folder and also add config file.

amarthadan commented on lib/cloudkeeper/aws/cli.rb:
```diff
@@ -0,0 +1,93 @@
+      SIGTERM = 15
+      SIGNALS = [SIGTERM, SIGINT].freeze
+
+      method_option :'polling-timeout',
```
Missing option description.

amarthadan commented on lib/cloudkeeper/aws/cli.rb:
```diff
@@ -0,0 +1,93 @@
+module Cloudkeeper
+  module Aws
+    # Class defining CLI of cloudkeeper-aws
+    class CLI < Thor
```
Missing options for logging and debugging. Is that on purpous? Will that be included in the logging PR?

amarthadan commented on lib/cloudkeeper/aws/cli.rb:
```diff
@@ -0,0 +1,93 @@
+
+      private
+
+      def initialize_cloud
```
What is the point of this method and keeping `Cloudkeeper::Aws::Cloud` as an instance variable? Wouldn't be easier to call `Cloudkeeper::Aws::CoreConnector.new(Cloudkeeper::Aws::Cloud.new)` directly?

amarthadan commented on lib/cloudkeeper/aws/cli.rb:
```diff
@@ -0,0 +1,93 @@
+      SIGTERM = 15
+      SIGNALS = [SIGTERM, SIGINT].freeze
+
+      method_option :'polling-timeout',
```
`cloudkeeper-one` is using gem `chronic_duration` to parse human readable time periods. Maybe you should considre doing the same.

dudoslav replied:
I considered it

amarthadan commented on lib/cloudkeeper/aws/cli.rb:
```diff
@@ -0,0 +1,93 @@
+                    default: Cloudkeeper::Aws::Settings['listen-address'],
+                    type: :string,
+                    desc: 'IP address gRPC server will listen on'
+      method_option :authentication,
```
You should add a check which will see whether when `authentication` option is set to `true` options `certificate`, `key` and `core-certificate` are set as well. This way `TypeError: no implicit conversion of nil into String` would be thrown and that is quite misleading.

Take a look at [https://github.com/the-cloudkeeper-project/cloudkeeper-one/blob/master/lib/cloudkeeper/one/cli.rb#L133](https://github.com/the-cloudkeeper-project/cloudkeeper-one/blob/master/lib/cloudkeeper/one/cli.rb#L133).

amarthadan commented on config/cloudkeeper-aws.yml:
```diff
@@ -1,5 +1,16 @@
+  level: ERROR # Logging level
+  file: /var/log/cloudkeeper/cloudkeeper-aws.log # File to write log to. To turn off file logging leave this field empty.
+debug: false # Debug mode
 aws:
```
What exactly can be set in this section? I thought that we would leave that for the AWS configuration file.

dudoslav committed:
Add CLI for cloudkeeper-aws

CLI includes method sync and version. Also this commit adds THOR into
dependencies, gem bin folder and also add config file.

amarthadan reviewed (approved):
:eagle:
